Elements/src/Algorithms/BinaryHeap.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
```

Comment:
Let's try to use existing PriorityQueue. I don't think we can have so similar classes.